examples/sensor-websocket/src/apiHandler.js

```diff
@@ -0,0 +1,203 @@
+import { reverse } from "named-urls";
+import axios from "axios";
+
+const { Socket } = require("phoenix");
+
+const ENDPOINT = {
+  device_alias: "devices-by-alias/:device_alias/",
+  device_id: "devices/:id?",
+  interface_by_alias: "devices/:device_alias/interfaces/:interface/",
+  interface_by_id: "devices/:device_id/interfaces/:interface/",
+  interface_id_path: "devices/:device_id/interfaces/:interface/:sensor_id/:key"
+};
+
+export default class ApiHandler {
+  constructor({ endpoint, realm, token, version = "v1" }) {
+    this.token = token;
+    this.endpoint = new URL(endpoint);
+    this.realm = realm;
+    this.version = version;
+    this.socket = null;
+  }
+
+  getDevice(device) {
+    if (this.isDeviceId(device)) {
+      return this.getDeviceDataById(device);
+    } else {
+      return this.getDeviceDataByAlias(device);
+    }
+  }
+
+  getHeaders() {
+    return {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${this.token}`
+    };
+  }
+
+  getAPIUrl(endPoint, params) {
+    const path = reverse(ENDPOINT[endPoint], params);
+    return new URL(
+      `/appengine/${this.version}/${this.realm}/${path}`,
+      this.endpoint
+    );
+  }
+
+  getWSUrl() {
+    const websocketUrl = new URL(
+      `/appengine/${this.version}/socket`,
+      this.endpoint
+    );
+    websocketUrl.protocol = "wss:";
+    return websocketUrl;
+  }
+
+  GET(url, params) {
+    return axios.get(url, { headers: this.getHeaders(), params: params });
+  }
+
+  PUT(url, params) {
+    return axios.put(url, params, { headers: this.getHeaders() });
+  }
+
+  isDeviceId(value) {
+    const expression = new RegExp(/^[a-zA-Z0-9-_]{22}$/g);
+    return value.length === 22 && expression.test(value)
```

Comment:
the previous regular expression already checks for length, no need to check it again